internal/handlers/common/compare.go

```diff
@@ -140,9 +164,29 @@ func compareScalars(a, b any) compareResult {
 		case int64:
 			return compareOrdered(a, b)
 		default:
-			panic(fmt.Sprintf("unexpected type %T", b))
+			return notEqual
 		}
 
+	case *types.Document:
+		return notEqual
+
+	case *types.Array:
```

Comment:
array and document should be there – they are not scalars

Reply:
array and document are not scalars so when compared with scalar types they return not equal